android-components/components/browser/engine-gecko/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngineSession.kt

```diff
@@ -1098,6 +1098,22 @@ class GeckoEngineSession(
         override fun onShowDynamicToolbar(geckoSession: GeckoSession) {
             notifyObservers { onShowDynamicToolbar() }
         }
+
+        override fun onGetNimbusFeature(session: GeckoSession, featureId: String): JSONObject? {
+            val json = JSONObject()
+            when (featureId) {
+                "pdfjs" -> {
+                    val pdfjs = GeckoNimbus.features.pdfjs.value()
+                    json.put("download-button", pdfjs.downloadButton)
+                    json.put("open-in-app-button", pdfjs.openInAppButton)
+                }
+                else -> {
+                    return null
+                }
+            }
+
+            return json
+        }
```

Comment:
Note to future: [this PR](https://github.com/mozilla/application-services/pull/5574) adds `getFeature(featureId: String)` and `toJSONObject()` for usage here.

Reply:
I updated the patch to use this amazing new feature: @jhugman thank you very much.